### PR TITLE
fix: Grid reactivity race condition

### LINKED ISSUE #8122: [spreadsheet] Fix GridSheet remounting issue

### PR DESCRIPTION
**Pull request by thure**
This PR:
- resolves #8122

https://github.com/user-attachments/assets/2623e52e-5f73-4c29-ae88-d69e1050ccd9

### CODE REVIEW

File: packages/plugins/plugin-sheet/src/components/GridSheet/GridSheet.tsx [hunk 1]
```diff
@@ -59,7 +59,7 @@ export const GridSheet = () => {
   const { t } = useTranslation(SHEET_PLUGIN);
   const { id, model, editing, setEditing, setCursor, setRange, cursor, cursorFallbackRange, activeRefs } =
     useSheetContext();
-  const dxGrid = useRef<DxGridElement | null>(null);
+  const [dxGrid, setDxGrid] = useState<DxGridElement | null>(null);
```
Comment (wittjosiah) [inline packages/plugins/plugin-sheet/src/components/GridSheet/GridSheet.tsx:62]: I'd suggest leaving a comment about why this needs to be state and not a ref

File: packages/ui/lit-grid/src/dx-grid.ts [hunk 2]
```diff
@@ -1496,8 +1496,14 @@ export class DxGrid extends LitElement {
     return false;
   }
 
+  override connectedCallback() {
+    super.connectedCallback();
+    console.log('[dx-grid connected]', this.gridId);
```
Comment (wittjosiah) [inline packages/ui/lit-grid/src/dx-grid.ts]: 👀

File: packages/ui/react-ui-grid/src/Grid/Grid.tsx [hunk 3]
```diff
@@ -106,22 +105,35 @@ const GRID_CONTENT_NAME = 'GridContent';
 
 const GridContent = forwardRef<NaturalDxGrid, GridScopedProps<GridContentProps>>((props, forwardedRef) => {
   const { id, editing, setEditBox, setEditing } = useGridContext(GRID_CONTENT_NAME, props.__gridScope);
-  const dxGrid = useForwardedRef(forwardedRef);
+  const [dxGrid, setDxGridInternal] = useState<NaturalDxGrid | null>(null);
+
+  const setDxGrid = useCallback(
```
Comment (wittjosiah) [inline packages/ui/react-ui-grid/src/Grid/Grid.tsx:110]: maybe similar note here

Comment (wittjosiah) [reply packages/ui/react-ui-grid/src/Grid/Grid.tsx:110]: todo: factor out hook?

Comment (thure) [reply packages/ui/react-ui-grid/src/Grid/Grid.tsx:110]: I‘m not sure that’s necessarily motivated this early, I can’t predict yet how frequently we’ll need this pattern. Will leave a note for sure though.

**Pull request merged**